### DISCUSSION TITLE
Best way to prevent deletion

### DISCUSSION THREAD

**Discussion by nrowealign** [jfrog/terraform-provider-artifactory#632] [Q&A] [answered]
We are using this provider purely for repository configuration which we've delegated to junior engineers. We'd like to ensure that repositories can't be deleted by mistake and must be deleted manually. We thought about using a user that lacked this permission, but it doesn't seem like it's possible to use a user other than `admin`.

Should I create a feature request that turns deletion off at the provider level or is there a better solution?

Thanks!

**Comment by alexhung** [comment, marked as answer]
@nrowealign There's no magic bullet to solve this problem. Authorization is not the responsibility of the provider so we will not implement any support. You should open a feature request ticket via JFrog's support to request the repository API be controlled by roles and/or permissions (vs user type only).

In addition my suggestion is to implement a combination of policy and practices with some technical hurdle to prevent accidental deletion.

For example:
1. Add `lifecycle.prevent_destroy` to the resources and set it to `true`. See [Terraform documentation](https://developer.hashicorp.com/terraform/tutorials/state/resource-lifecycle#prevent-resource-deletion).
2. Use [gitops](https://about.gitlab.com/topics/gitops/) process and implement PR/code review, as the `lifecycle.prevent_destroy` must be modified in order to perform the deletion.
3. Use CI/CD pipeline and implement manual approval step(s), based on output from a Terraform dry run.

**Reply by nrowealign** [reply]
Thanks, Alex!
